nixos/modules/installer/tools/tools.nix

```diff
@@ -10,236 +10,292 @@ let
+        Disable nixos-rebuild, nixos-generate-config, nixos-installer
+        and other NixOS tools. This is useful to shrink embedded,
+        read-only systems which are not expected to be rebuild or
+        reconfigure themselves. Use at your own risk!
       '';
     };
 
-    desktopConfiguration = mkOption {
-      internal = true;
-      type = types.listOf types.lines;
-      default = [];
-      description = lib.mdDoc ''
-        Text to preseed the desktop configuration that `nixos-generate-config`
-        saves to `/etc/nixos/configuration.nix`.
-
-        This is an internal option. No backward compatibility is guaranteed.
-        Use at your own risk!
-
-        Note that this string gets spliced into a Perl script. The perl
-        variable `$bootLoaderConfig` can be used to
-        splice in the boot loader configuration.
-      '';
+    system.installerTools = {
+      nixos-version.selfContained = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = lib.mdDoc ''
+          Whether nixos-version is a self-contained script with a static reference
+          to the info it provides.
+          The alternative is to always read this info from /run/current-system.
+
+          The reason for not making nixos-version self-contained, is that doing so
+          causes a lot of rebuilds whenever the info contained in the JSON file,
+          like the nixpkgs revision hash, changes.
+        '';
+      };
+
+      tools = lib.mkOption {
+        internal = true;
+        type = lib.types.attrsOf (lib.types.submodule ({ name, config, lib, ... }: {
+          options = {
+            name = lib.mkOption {
+              type = lib.types.str;
+              description = lib.mdDoc ''
+                the name of the package
+              '';
+            };
+
+            enable = lib.mkEnableOption (lib.mdDoc "the package as part of the installer tools.");
+
+            package = lib.mkOption {
+              type = lib.types.package;
+              description = lib.mdDoc ''
+                the derivation providing the package
+              '';
+            };
+          };
+
+          config.name = lib.mkDefault name;
+        }));
+      };
     };
   };
 
-  options.system.disableInstallerTools = mkOption {
-    internal = true;
-    type = types.bool;
-    default = false;
-    description = lib.mdDoc ''
-      Disable nixos-rebuild, nixos-generate-config, nixos-installer
-      and other NixOS tools. This is useful to shrink embedded,
-      read-only systems which are not expected to be rebuild or
-      reconfigure themselves. Use at your own risk!
-    '';
-  };
+  config =
+    let
+      installerTools = lib.filterAttrs (_: tool: tool.enable) config.system.installerTools.tools;
+    in
+    lib.mkIf (config.nix.enable && !config.system.disableInstallerTools) {
+
+      system.installerTools.tools = {
+        nixos-build-vms = {
+          enable = true;
```

Comment:
Why not `programs.nixos-build-vms`?
These are perfectly normal "user" commands that you might as well get from `nix shell` are whatever; no need to treat categorize them differently.